src/journal/compress.c

```diff
@@ -49,15 +58,36 @@ DEFINE_TRIVIAL_CLEANUP_FUNC(LZ4F_compressionContext_t, LZ4F_freeCompressionConte
 DEFINE_TRIVIAL_CLEANUP_FUNC(LZ4F_decompressionContext_t, LZ4F_freeDecompressionContext);
 #endif
 
+#if HAVE_ZSTD
+DEFINE_TRIVIAL_CLEANUP_FUNC(ZSTD_CCtx*, ZSTD_freeCCtx);
+DEFINE_TRIVIAL_CLEANUP_FUNC(ZSTD_DCtx*, ZSTD_freeDCtx);
+#endif
+
 #define ALIGN_8(l) ALIGN_TO(l, sizeof(size_t))
 
 static const char* const object_compressed_table[_OBJECT_COMPRESSED_MAX] = {
         [OBJECT_COMPRESSED_XZ] = "XZ",
         [OBJECT_COMPRESSED_LZ4] = "LZ4",
+        [OBJECT_COMPRESSED_ZSTD] = "ZSTD",
 };
 
 DEFINE_STRING_TABLE_LOOKUP(object_compressed, int);
 
+#if HAVE_ZSTD
+static int zstd_ret_to_errnoh(size_t ret) {
+        ZSTD_ErrorCode error = ZSTD_getErrorCode(ret);
+
+        switch (error) {
+        case ZSTD_error_dstSize_tooSmall:
+                return -EFBIG;
```

Comment:
Do you do anything special if `-EFBIG` is the error code?

Reply:
No

Reply:
Ok, then it should be okay to inspect the error codes, since the worst case is reporting the wrong error.

Since the error codes were only stabilized in v1.3.1 , If you want, you could check if the version number is `ZSTD_VERSION_NUMBER >= 10301`, and only report specific error codes  if it is true.